docs/DeveloperGuide.md

```diff
@@ -6,33 +6,183 @@
+| Version | As a ...                 | I want to ...                                 | So that I can ...                                    |
+|---------|--------------------------|-----------------------------------------------|------------------------------------------------------|
+| v1.0    | student                  | focus on financial responsibility             | view a breakdown of his expenditures by category     |
+| v1.0    | Proactive business major | calculate my course grades on CantVas         | see which mod I have to SU.                          |
+| v1.0    | careless user            | delete wrong items in the expenditure tracker | add back the correct item in the expenditure tracker |
 
 ## Non-Functional Requirements
 
 {Give non-functional requirements}
 
+## Environment Setup
+Java JDK 8 or above
+An IDE that supports Java (e.g., IntelliJ IDEA, Eclipse)
+
 ## Glossary
 
 * *glossary item* - Definition
 
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+# Motivational messages
+The Motivational module prints a randomly generated motivational quote for the user whenever the program starts 
+
+# Expenditure
+The expenditure module is comprehensive allows users to key in their expenses to enable them to store all their 
+
+## Overview
+The expenditure module allows users to key in their expenses to enable them to store all their 
+expenses.
+This guide will walk developers through how to use the expenditure function to track their expenses
+
+### List of Operations:
+#### Deleting an expenditure:
+Deleting an expenditure in numerical form, by referencing its index in the array.
+Format: e/ del/ INDEX
+Example: e/ del/ 3
+#### Adding an expenditure:
+Adding an expenditure in numerical form.
+Format: add n/DOLLARS_AND_CENTS d/DD.MM.YYYY
+Example: add n/3.22 d/31.01.2024
+#### Viewing expenditures by month:
+Viewing an expenditure in numerical form, filtered by month.
+Format: view -m MM.YYYY
+Example: view -m 01.2023
+#### View expenditure by year:
+Viewing an expenditure in numerical form, filtered by year.
+Format: view -y YEAR
+Example: view -y 2023
+#### Viewing expenditures:
+Viewing an expenditure in numerical form, in sequence of when they are added.
+Format: list
+Example: list
+
+
+
+# GPA Function
+The GPA Calculator Module is a comprehensive component designed to facilitate the calculation 
+of a user's updated GPA based on their current academic standing and projected module grades. 
+This guide will walk developers through the architecture, functionality, and core components of the module.
+
+------------------------------------------------------------------------------------------
+
+## Overview
+The module is divided into two primary classes:
+
+------------------------------------------------------------------------------------------
+
+## Class: Expenditure
```

Comment:
Would be good to add class diagrams to represent how your various classes interact with one another